### PR TITLE
Support for creating and signing quote bundles

### PR DESCRIPTION
**Pull request by mightyost**
# Jira Issue: [IPC-34, IPC-35] 

## What?
Added two new endpoints for creating and signing quote bundles:

```
  @PostMapping("bundle")
  @Secured("ROLE_COMPARISON")
  fun bundleQuotes(
    @Valid @RequestBody request: BundleQuotesRequestDTO
  ): ResponseEntity<out Any>
  
  @PostMapping("bundle/sign")
  @Secured("ROLE_COMPARISON")
  fun signBundle(
    @Valid @RequestBody request: SignBundleRequestDTO
  ): ResponseEntity<out Any>
```

## Why?
Needed for Norwegian partners

## Optional checklist
- [x] Boyscouted
- [x] Unit tests written
- [x] Tested locally

### CODE REVIEW

File: src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt [hunk 1]
```diff
@@ -30,12 +31,27 @@ class ConcreteUnderwriter(private val client: UnderwriterClient,
...
+        }
+    }
+
+    override fun quoteBundle(request: QuoteBundleRequestDto): Either<ErrorResponse, QuoteBundleResponseDto> {
+        try {
+            val response = this.client.quoteBundle(request)
+            return Either.right(response.body!!)
+        } catch (ex: Exception) {
+            logger.warn { "Failed to get quote bundle from Underwriter: $ex" }
+            throw RuntimeException("Underwriter error")
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt:54]:
```suggestion
            throw ex
```

File: src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt [hunk 2]
```diff
@@ -44,12 +60,44 @@ class ConcreteUnderwriter(private val client: UnderwriterClient,
...
+            logger.warn { "Failed to sign quote from Underwriter: $ex" }
             if (ex.status() == 422) {
                 val error = objectMapper.readValue<ErrorResponse>(ex.contentUTF8())
                 return Either.left(error)
             }
-
             throw ex
+        } catch (ex: Exception) {
+            logger.error { "Failed to sign quote from Underwriter: $ex" }
+            throw RuntimeException("Underwriter error")
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt:71]:
```suggestion
            throw ex
```

File: src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt [hunk 3]
```diff
@@ -44,12 +60,44 @@ class ConcreteUnderwriter(private val client: UnderwriterClient,
...
+            logger.warn { "Failed to get quote bundle from Underwriter: $ex" }
+            if (ex.status() == 422) {
+                val error = objectMapper.readValue<ErrorResponse>(ex.contentUTF8())
+                return Either.left(error)
+            }
+            throw RuntimeException("Underwriter error")
+
+        } catch (ex: Exception) {
+            logger.error { "Failed to get quote bundle from Underwriter: $ex" }
+            throw RuntimeException("Underwriter error")
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt:99]:
```suggestion
            throw ex
```

File: src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt [hunk 4]
```diff
@@ -44,12 +60,44 @@ class ConcreteUnderwriter(private val client: UnderwriterClient,
...
+        try {
+            val response = client.signQuoteBundle(SignQuoteBundleRequest(ids, SignQuoteBundleRequest.Name(firstName, lastName), ssn, startsAt, email, price, currency))
+            return Either.right(response.body!!)
+        } catch (ex: FeignException) {
+            logger.warn { "Failed to get quote bundle from Underwriter: $ex" }
+            if (ex.status() == 422) {
+                val error = objectMapper.readValue<ErrorResponse>(ex.contentUTF8())
+                return Either.left(error)
+            }
+            throw RuntimeException("Underwriter error")
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt:95]:
```suggestion
            throw ex
```

**Review by FredrikAreschoug — COMMENTED** [review]
Looks good! Added some comments on throwing `RuntimeException` Why I don't like it is mostly because tools like Sentry. Will group all of these errors when they are not really the same thing. Or maybe we should return `Left.of(error)`

File: src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt (hunk 1 above)
Comment (mightyost) [reply src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt:54]: Hmm, in this case I think it is an error? I may be wrong but I guess we will get a `FeignException` in more or less all cases even if underwriter responses with 500 or times out whatever, but I am just guessing?

Comment (mightyost) [reply src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/ConcreteUnderwriter.kt:54]: Ah, I see what u mean in this case :)

**Review by FredrikAreschoug — APPROVED** [review]

**Pull request merged**